code/game/objects/items/crayons.dm

```diff
@@ -123,17 +128,59 @@ var/global/list/all_graffitis = list(
 				if(!preference) return
 
 				drawtype=graffitis[preference]
-				to_chat(user, "You start drawing graffiti on the [target.name].")
+				to_chat(user, "You start drawing graffiti on \the [target].")
 			if("rune")
-				to_chat(user, "You start drawing a rune on the [target.name].")
+				to_chat(user, "You start drawing a rune on \the [target].")
+			if("text")
+				#define MAX_LETTERS 15
+				preference = input("Write some text here (maximum [MAX_LETTERS] letters).", "Crayon scribbles") as null|text
+				preference = copytext(preference, 1, MAX_LETTERS)
+				#undef MAX_LETTERS
+
+				var/letter_amount = length(replacetext(preference, " ", ""))
+				if(!letter_amount) //If there is no text
+					return
+				drawtime = 4 * letter_amount //10 letters = 4 seconds
+
+				alignment = input("Select vertical text alignment (your text is \"[preference]\")", "Crayon scribbles") as null|anything in list("middle", "bottom", "top")
+				if(!alignment) return
+
+				if(user.client)
+					var/image/I = image(icon = null)
+					I.maptext = {"<span style="color:[colour];font-size:[FONT_SIZE];font-family:'[FONT_NAME]';" valign="[alignment]">[preference]</span>"}
+					I.loc = get_turf(target)
+					animate(I, alpha = 100, 10, -1)
+					animate(alpha = 255, 10, -1)
+
+					user.client.images.Add(I)
+					var/continue_drawing = alert(user, "This is how your drawing will look. Continue?", "Crayon scribbles", "Yes", "Cancel")
+
+					user.client.images.Remove(I)
+					qdel(I)
```

Comment:
You can't actually qdel() an image. You just remove all the references to it, as well as cancel the animation in this case by doing animate(whatever).